src/agentscope/agents/dialog_agent.py

```diff
@@ -68,13 +68,13 @@ def reply(self, x: dict = None) -> dict:
             response to the user's input.
         """
         # record the input if needed
-        if x is not None:
+        if self.memory:
```

Comment:
If set `use_memory=False` when initializing an agent, `self.memory` is not defined, which will cause `AttributeError: 'DialogAgent' object has no attribute 'memory'`

Reply:
The bug is fixed accordingly.